app/models/concerns/sluggable.rb

```diff
@@ -9,6 +9,7 @@ module Sluggable
   end
 
   def slugify
+    return if slug.present?
     self.slug = title.parameterize
   end
```

Comment:
I think the slug should be present in the form, and we should require it for Assignment creation.